enhancements/kubelet/virtualization-higher-workload-density.md

```diff
@@ -0,0 +1,474 @@
+
+Now - Extrapolating this behavior from one to all virtual machines on a
+given node leads to the observation that _on average_ much of the reserved memory is not utilized.
+Moreover, the memory pages that are utilized can be classified to a frequently used (a.k.a working set) and inactive memory pages.
+In case of memory pressure the inactive memory pages can be swapped out. 
+From the cluster owner perspective, reserved but underutilized hardware resources - like memory in this case -
+are a cost factor.
+
+This proposal is about increasing the virtual machine density and thus
+memory utilization per node, in order to reduce the cost per virtual machine.
+
+> [!NOTE]
+> CPU resources are already over-committed in OpenShift today, thus the
+> limiting factor - usually - are memory resources.
+
+### User Stories
+
+* As an administrator, I want to be able to enable higher workload
+  density for OpenShift Virtualization clusters so that I can increase
+  the memory utilization.
+
+### Goals
+
+#### Functionality
+
+* Fit more virtual machines onto a node once higher workload density
+  is enabled
+* Integrate well with [Kernel Samepage Merging] and [Free Page Reporting]
+* Protect the node from resource starvation upon high swapping activity.        
+
+#### Usability
+
+* Provide a boolean in the OpenShift Console for enabling higher-density.
+  Do not require any additional workload or cluster level configuration
+  besides the initial provisioning.
+* Provide a method to deploy the requried configuration for safer
+  memory overcommit, such as swap provisioning, tuning of system services, etc.
+* Provide a maintainable delivery method for new releases of the proposal.
+
+### Non-Goals
+
+* Allow swapping for VM pods only. We don't want to diverge from the upstream approach 
+  since [Kubernetes SWAP] allows swapping for all pods associated with the burtsable QoS class.
+* Swap Operator graduation. This will be discussed in a dedicated proposal.
+* Allow choosing how to enable swap for pods: wasp-agent or kube swap. That is an implementation detail, hence it should be 
+  transparent to the end user.
+* SNO: MicroShift and Compact Cluster deployments are out of scope of this
+  proposal, since enabling swap on control-plane nodes is not supported.
+* HCP: Hosted cluster with OCP Virt, since nested virt is not supported.
+
+## Proposal
+
+A higher workload density is achieved by combining two mechanisms
+
+1. Under-request memory resources for virtual machines by having the VM
+   pods (`launcher`) `requests.memory` being smaller than the VMs `guest.memory`
+   according to a configured ratio[^1]. This is owned by
+   [KubeVirt and present today](https://kubevirt.io/user-guide/operations/node_overcommit/).
+2. Compensate for the memory over-committment by using SWAP in order
+   to extend the virtual memory. This is owned by the platform and not
+   available today.
+
+
+### Applicability
```

Comment:
This section looks like it is covered under goals and could either be folded up there.